### PR TITLE
Bugfix/date picker mask weird behavior + inconsistent date picker locale in Showcase and Starter

### PR DESCRIPTION
**Pull request by christophercr**
ISSUES CLOSED: #1260 

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/NationalBankBelgium/stark/blob/master/CONTRIBUTING.md#-commit-message-guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[X] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #1260

## What is the new behavior?
- now when deleting a character, the mask keeps the position as empty instead of moving the characters backwards so the user can safely type in the new character. This prevents the date picker from getting weird date values. The `keepCharPositions` option is now enabled in the timestamp directive (see https://github.com/text-mask/text-mask/blob/master/componentDocumentation.md#keepcharpositions)
- the max and min range values for the different date parts of a timestamp are now correct.
- expanded the unit tests of the `createTimestampPipe` to cover all possible use cases with date time strings.
- enhanced the translation initialization to also initialize the Moment locale to keep it in sync with the application language.
- implemented some logic to change the Moment locale and the Material Date Adapter when the app language changes.

**IMPORTANT: Although the locale of the Material Date Adapter is now changed when the app language changes, the date pickers don't reflect the new locale. This is a known bug in Angular Material (see https://github.com/angular/components/issues/15419) and should be fixed soon.**

**So for the moment, the logic has been put in the `SharedModule` instead to make it work**

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

### CODE REVIEW

**Comment by christophercr** [comment]
Could you have a look on this PR?

File: showcase/src/app/app.module.ts [hunk 1]
```diff
@@ -263,12 +266,33 @@ export class AppModule {
...
 			.pipe(filter((user?: StarkUser): user is StarkUser => typeof user !== "undefined"))
 			.subscribe((user: StarkUser) => {
 				const devAuthenticationHeaders: Map<string, string> = getAuthenticationHeaders(user);
 				this.sessionService.setDevAuthenticationHeaders(devAuthenticationHeaders);
 			});
+
+		/**
+		 * When the application language changes, the Moment locale and the Material Moment Adapter (internally used by Stark Date Pickers) should be changed as well
+		 */
+		this.sessionService
```
Comment (SuperITMan) [inline showcase/src/app/app.module.ts]: It's unfortunately not working when it's part of the **App** module. Could you please move this logic in the **Shared** module to make it work ? 😊

File: starter/src/app/app.module.ts [hunk 2]
```diff
@@ -218,12 +221,33 @@ export class AppModule {
...
 			.pipe(filter((user?: StarkUser): user is StarkUser => typeof user !== "undefined"))
 			.subscribe((user: StarkUser) => {
 				const devAuthenticationHeaders: Map<string, string> = getAuthenticationHeaders(user);
 				this.sessionService.setDevAuthenticationHeaders(devAuthenticationHeaders);
 			});
+
+		/**
+		 * When the application language changes, the Moment locale and the Material Moment Adapter (internally used by Stark Date Pickers) should be changed as well
+		 */
+		this.sessionService
```
Comment (SuperITMan) [inline starter/src/app/app.module.ts]: Same here about moving this in the **Shared** module

File: packages/stark-ui/src/modules/input-mask-directives/directives/timestamp-mask.directive.ts [hunk 3]
```diff
@@ -122,7 +122,8 @@ export class StarkTimestampMaskDirective extends MaskedInputDirective implements
 			return {
 				pipe: <any>createTimestampPipe(timestampMaskConfig.format),
 				mask: this.convertFormatIntoMask(timestampMaskConfig.format),
-				placeholderChar: "_"
+				placeholderChar: "_",
+				keepCharPositions: true // to avoid weird date values when deleting characters (see https://github.com/NationalBankBelgium/stark/issues/1260)
```
Comment (SuperITMan) [inline packages/stark-ui/src/modules/input-mask-directives/directives/timestamp-mask.directive.ts]: It's not really working fine... If we have 13/05/2019 and we want to change the month for December, we cannot do it.
We can obtain 13/0_/2019 but not 13/__/2019... Maybe it's already better than nothing 😊

Comment (christophercr) [reply packages/stark-ui/src/modules/input-mask-directives/directives/timestamp-mask.directive.ts]: As discussed, I've partially fixed this issue. There are still some edge cases that are not covered but I think they are not urgent for now so we can release soon.

So I've created #1277 to track the issue for those edge cases that we should fix later on 😉

**Comment by christophercr** [comment]
@SuperITMan PR updated with the remarks. Can you review them?

**Pull request merged**